torchvision/datasets/core50.py

```diff
@@ -0,0 +1,229 @@
+        check_integrity (bool, optional): If True check the integrity of the
+            Dataset before trying to load it.
+        scenario (string, optional): One of the three scenarios of the CORe50
+            benchmark ``ni``, ``nc`` or ``nic``.
+        train (bool, optional): If True, creates the dataset from the training
+            set, otherwise creates from test set.
+        img_size (string, optional): One of the two img sizes available among
+            ``128x128`` or ``350x350``.
+        cumul (bool, optional): If True the cumulative scenario is assumed, the
+            incremental scenario otherwise. Practically speaking ``cumul=True``
+            means that for batch=i also batch=0,...i-1 will be added to the
+            available training data.
+        run (int, optional): One of the 10 runs (from 0 to 9) in which the
+            training batch order is changed as in the official benchmark.
+        batch (int, optional): One of the training incremental batches from 0 to
+            max-batch - 1. Remember that for the ``ni``, ``nc`` and ``nic`` we
+            have respectively 8, 9 and 79 incremental batches. If
+            ``train=False`` this parameter will be ignored.
+        transform (callable, optional): A function/transform that takes in an
+            PIL image and returns a transformed version. E.g,
+            ``transforms.ToTensor()``
+        target_transform (callable, optional): A function/transform that takes
+            in the target and transforms it.
+        download (bool, optional): If true, downloads the dataset from the
+            internet and puts it in root directory. If dataset is already
+            downloaded, it is not downloaded again.
+
+    Example:
+
+        .. code:: python
+
+            training_data = datasets.CORE50(
+                '~/data/core50', transform=transforms.ToTensor(), download=True
+            )
+            training_loader = torch.utils.data.DataLoader(
+                training_data, batch_size=128, shuffle=True, num_workers=4
+            )
+            test_data = datasets.CORE50(
+                '~/data/core50', transform=transforms.ToTensor(), train=False,
+                download=True
+            )
+            test_loader = torch.utils.data.DataLoader(
+                training_data, batch_size=128, shuffle=True, num_workers=4
+            )
+
+            for batch in training_loader:
+                imgs, labels = batch
+                ...
+
+        This is the simplest way of using the Dataset with the common Train/Test
+        split. If you want to use the benchmark as in the original CORe50 paper
+        (that is for continuous learning) you need to play with the parameters
+        ``scenario``, ``cumul``, ``run`` and ``batch`` hence creating a number
+        of Dataset objects (one for each incremental training batch and one for
+        the test set).
+
+    """
+    ntrain_batch = {
```

Comment:
This is a bit confusing, as it's not used (with this name) anywhere in the code, and is referenced as `max-batch` in the documentation.

If you want to keep this around, I think you should add some asserts in the `__init__` checking that `batch` is within `ntrain_batch`